src/Business Foundation/App/NoSeries/src/Single/NoSeriesStatelessImpl.Codeunit.al

```diff
@@ -28,6 +28,11 @@ codeunit 306 "No. Series - Stateless Impl." implements "No. Series - Single"
 
     procedure GetNextNo(var NoSeriesLine: Record "No. Series Line"; UsageDate: Date; HideErrorsAndWarnings: Boolean): Code[20]
     begin
+        if not NoSeriesLine.IsTemporary() then begin
+            NoSeriesLine.LockTable();
```

Comment:
Fine, but I would go for a more modern approach and write NoSeriesLine.ReadIsolation(IsolationLevel::UpdLock).

Also, I think this part goes together with the 'if ModifySeries then ....' section in GetNextNoInternal, so either the update is moved into this function or the lock+read is moved to GetNextNoInternal.

Reply:
The same locking is also missing in NoSeriesSequenceImpl.